### PR TITLE
[indexer] Optimize RulesHolder

### PR DESCRIPTION
**Pull request by pastk**
Simplified/optimized how drules are indexed. Now fetching them will be faster.

Also deleted some obsolete unused code.

### CODE REVIEW

**Review by vng — APPROVED** [review]

File: indexer/drawing_rule_def.hpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ int32_t constexpr kOverlaysMaxPriority = 10000;
```
Comment (vng) [inline indexer/drawing_rule_def.hpp:29]: nit: size_t i

File: indexer/drawing_rule_def.hpp [hunk 2]
```diff
@@ -20,7 +20,7 @@ int32_t constexpr kOverlaysMaxPriority = 10000;
   public:
     int m_scale = -1;
     int m_type = -1;
-    int m_index = -1;
+    size_t m_index = std::numeric_limits<size_t>::max(); // an index to RulesHolder.m_dRules[]
```
Comment (vng) [inline indexer/drawing_rule_def.hpp:23]: I think, def initialization is useless here, can remove.

Comment (pastk) [reply indexer/drawing_rule_def.hpp:23]: In many places Key is created via a default constructor, which doesn't assign anything.
So theoretically its possible it'll be passed undefined into e.g. RulesHolder::Find(). Am I right?

File: indexer/drawing_rule_def.hpp (hunk 1 above)
Comment (pastk) [reply indexer/drawing_rule_def.hpp:29]: fixed, thanks!

File: indexer/drawing_rule_def.hpp (hunk 2 above)
Comment (vng) [reply indexer/drawing_rule_def.hpp:23]: Yes, no problem here. Anyway, max() is the same as 0 is the same as nothing.
def ctor needed for buffer_vector

**Review by vng — APPROVED** [review]

**Review by vng — APPROVED** [review]

**Pull request merged**